Xamarin.PropertyEditing.Tests/ThicknessPropertyViewModelTests.cs

```diff
@@ -0,0 +1,152 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using NUnit.Framework;
+using Xamarin.PropertyEditing.Drawing;
+using Xamarin.PropertyEditing.ViewModels;
+
+namespace Xamarin.PropertyEditing.Tests
+{
+	[TestFixture]
+	internal class ThicknessPropertyViewModelTests
+		:PropertyViewModelTests<CommonThickness, ThicknessPropertyViewModel>
+	{
+		[Test]
+		public void Top ()
+		{
+			var property = GetPropertyMock ();
+			var editor = GetBasicEditor (property.Object);
+			var vm = GetViewModel (property.Object, new [] { editor });
+			Assume.That (vm.Value, Is.EqualTo (new CommonThickness (0, 0, 0, 0)));
+
+			bool topChanged = false, valueChanged = false;
+			vm.PropertyChanged += (sender, args) => {
+				if (args.PropertyName == nameof (ThicknessPropertyViewModel.Top))
+					topChanged = true;
+				if (args.PropertyName == nameof (ThicknessPropertyViewModel.Value))
+					valueChanged = true;
+			};
+
+			vm.Top = 5;
+			Assert.That (vm.Value.Top, Is.EqualTo (5));
```

Comment:
Note: notifications for all four properties, plus Value, will be triggered in all these cases. Not sure if we care.